src/tag/tag-filter.component.ts

```diff
@@ -2,37 +2,59 @@ import {
 	Component,
 	Output,
 	EventEmitter,
-	HostBinding
+	HostBinding,
+	Input,
+	ViewChild,
+	ElementRef
 } from "@angular/core";
 import { Tag } from "./tag.component";
 
 @Component({
 	selector: "ibm-tag-filter",
 	template: `
-		<ng-content></ng-content>
-		<svg
-			(click)="close.emit()"
-			focusable="false"
-			preserveAspectRatio="xMidYMid meet"
+		<span #label class="bx--tag__label" [title]="label.innerHTML">
+			<ng-content></ng-content>
+		</span>
+		<button
 			class="bx--tag__close-icon"
-			style="will-change: transform;"
-			xmlns="http://www.w3.org/2000/svg"
-			aria-label="Clear filter"
-			width="16"
-			height="16"
-			viewBox="0 0 16 16"
-			role="img">
-			<path d="M12 4.7l-.7-.7L8 7.3 4.7 4l-.7.7L7.3 8 4 11.3l.7.7L8 8.7l3.3 3.3.7-.7L8.7 8z"></path>
-		</svg>
+			(click)="close.emit()"
+			[disabled]="disabled"
+			[attr.aria-labelledby]="id || tagId"
+			[title]="title">
+			<span class="bx--visually-hidden">{{title}}</span>
+			<svg ibmIconClose size="16"></svg>
+		</button>
 	`
 })
 export class TagFilter extends Tag {
+	static tagFilterCount = 0;
+	@Input() tagId = `tag-filter-${TagFilter.tagFilterCount}`;
+	@Input() title = "Clear Filter";
+	@Input() id: string;
+	@Input() disabled: boolean;
+
 	/**
 	 * Function for close/delete the tag
 	 */
 	@Output() close = new EventEmitter<any>();
 
+	@ViewChild("label") label: ElementRef;
+
 	@HostBinding("attr.class") get attrClass() {
 		return `bx--tag bx--tag--filter bx--tag--${this.type} ${this.class}`;
 	}
+
+	@HostBinding("attr.id") get attrId() {
+		return this.id ? this.id : this.tagId;
+	}
+
+	@HostBinding("attr.aria-label") get attrAriaLabel() {
+		const label = this.label.nativeElement.innerHTML;
+		return `${this.title} ${label}`;
```

Comment:
This may end up adding a lot of HTML junk to the `aria-label` ... I do wonder if this is something we should provide? Maybe it'll be clearer when we sort out the title and label situation

Reply:
Precisely, if we change the title and label situation to an Input, this will be much cleaner.